cdap-ui/app/directives/my-pipeline-resource/my-pipeline-executor-resource-ctrl.js

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+*/
+function MyPipelineExecutorResourceCtrl($scope) {
```

Comment:
So its essentially two directives but sharing the same template. The template is same. The controller needs to update different properties in the state.